### PR TITLE
get Database from DI in Library helper

### PR DESCRIPTION
**Pull request by sergeytolkachyov**
Update code base from old to new sintax

### Summary of Changes

`Factory::getDbo` changed to `Factory::getContainer()->get(DatabaseInterface::class)`

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by joomdonation** [comment]
@sergeytolkachyov Could not say that I understand these container stuffs, however, base on what I see from core code and by reading comments from maintainers on other similar PRs in the past, you should use `Factory::getContainer()->get(DatabaseInterface::class)` instead of using `Factory::getContainer()->get('DatabaseDriver')` as how you are doing here. See https://github.com/joomla/joomla-cms/blob/4.2-dev/modules/mod_finder/src/Helper/FinderHelper.php#L82 for an example

**Comment by sergeytolkachyov** [comment]
> @sergeytolkachyov Could not say that I understand these container stuffs, however, base on what I see from core code and by reading comments from maintainers on other similar PRs in the past, you should use `Factory::getContainer()->get(DatabaseInterface::class)` instead of using `Factory::getContainer()->get('DatabaseDriver')` as how you are doing here. See https://github.com/joomla/joomla-cms/blob/4.2-dev/modules/mod_finder/src/Helper/FinderHelper.php#L82 for an example

Thank you, I changed  it

File: libraries/src/Helper/LibraryHelper.php [hunk 1]
```diff
@@ -16,6 +16,7 @@
 use Joomla\CMS\Language\Text;
 use Joomla\CMS\Log\Log;
 use Joomla\Registry\Registry;
+use Joomla\Database\DatabaseInterface;
```
Comment (Quy) [inline libraries/src/Helper/LibraryHelper.php]:
```suggestion
use Joomla\Database\DatabaseInterface;
use Joomla\Registry\Registry;
```

**Review by laoneo — APPROVED** [review]

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on d9dbca328b1eba6fb239f747743e3932ef3f961a<br /><br />Both code review and run some code to test the methods which are affected by this change.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39276">issues.joomla.org/tracker/joomla-cms/39276</a>.</sub>

**Comment by joomdonation** [comment]
If someone wants to do real test, add this line of code below to index.php file of your template, then access to backend and see data for the phpass lib displayed:

```php
$lib = \Joomla\CMS\Helper\LibraryHelper::getLibrary('phpass');
var_dump($lib);
```

I also tested saveParams method, too. So maybe you just need to do a code review should be enough.

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on d9dbca328b1eba6fb239f747743e3932ef3f961a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39276">issues.joomla.org/tracker/joomla-cms/39276</a>.</sub>

**Comment by alikon** [comment]
rtc

**Comment by richard67** [comment]
RTC this time with the issue tracker.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39276">issues.joomla.org/tracker/joomla-cms/39276</a>.</sub>

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**